File: save_fcn.py
import os

def save_function(mode):
    # Placeholder function for saving data
    print(f"Saving data with mode {mode}...")

    # Example: Simulate file saving
    file_path = os.path.join("..", "Gamedata", "saved_file.saf")

    if mode == 0:
        print(f"Saving to existing file: {file_path}")
    elif mode == 1:
        print(f"Saving all data to a new file: {file_path}")
    elif mode == 2:
        print(f"Saving selected data to a new file: {file_path}")
    else:
        print("Invalid save mode.")
        return

    # Simulate saving the file
    print(f"Data successfully saved to {file_path}")

File: SaveFcn.m
function SaveFcn(h, event, mode)
global SA


file0 = '';%fullfile(SA.fullfilename);

switch mode
   case 0
      if exist(file0, 'file') ~= 2
         [FileName,PathName,FilterIndex] = uiputfile('*.saf','Save Shot Analyzer File', file0);
         if isnumeric(FileName)
            disp('abort')
            return
         end
         file = [PathName, FileName];
         
      else
         
         file = file0;
      end
      
      SA1 = SA;

   case 1
      % Save all in to new file
      
      [FileName,PathName,FilterIndex] = uiputfile('*.saf','Save Shot Analyzer File', file0);
      if isnumeric(FileName)
         disp('abort')
         return
      end
      
      file = [PathName, FileName];
      SA1 = SA;
      
      %for i = 1:length(SA.Shot)
      %    SA1.Shot(i) = rmfield(SA1.Shot,'Route');
      %end

    case 2
      % save selected in to new file
      [FileName,PathName,FilterIndex] = uiputfile('*.saf','Save Shot Analyzer File', file0);
      if isnumeric(FileName)
         disp('abort')
         return
      end
      
      file = [PathName, FileName];
      
      ind = cell2mat(SA.Table.Selected);

      SA1.Shot = SA.Shot(ind);
      SA1.Table = SA.Table(ind,:);
   
end

disp('Start saving ...')
save_shot(file, SA1);

SA.fullfilename = file;

disp('done with save')

function save_shot(file, SA)
save(file, 'SA','-v7.3');
